api/v3/TwingleDonation/Submit.php

```diff
@@ -484,8 +484,28 @@ function civicrm_api3_twingle_donation_Submit($params) {
       $result_values['organization'] = $organisation_id;
     }
 
+    // If Twingle's Double-Opt-In procedure is used, add contact with status "pending" to the newsletter groups
+    // defined in the profile
+    if (!empty($profile->getAttribute('double_opt_in')) &&
+        !empty($params['newsletter']) &&
+        !empty($groups = $profile->getAttribute('newsletter_groups'))) {
+        foreach ($groups as $group_id) {
+            if (empty(civicrm_api3('GroupContact', 'get', array(
+                'group_id' => $group_id,
+                'contact_id' => $contact_id,
+                ))['values'])) {
+                  civicrm_api3('GroupContact', 'create', array(
+                    'group_id' => $group_id,
+                    'contact_id' => $contact_id,
+                    'status' => "Pending",
+                ));
+                $result_values['double_opt_in'][] = "Pending";
+            }
```

Comment:
Can we make the `GroupContact.get` a `GroupContact.getsingle` wrapped with a `try` and put the `GroupContact.create` in the `catch`? This would avoid "Undefined index" warnings when something goes wrong with the API call. There can only be one `GroupContact` record per `contact_id`-`group_id` combination.

Reply:
Good idea!